src/app.js

```diff
@@ -654,23 +668,28 @@ var run = function () {
 
   function runCompiler () {
     var files = {}
-    var target = currentFile
-
-    files[target] = editor.get(currentFile)
-
-    compiler.compile(files, target)
+    if (currentFile) {
+      var target = currentFile
+      files[target] = editor.get(currentFile)
+      compiler.compile(files, target)
+    }
   }
 
   function editorSyncFile () {
-    var input = editor.get(currentFile)
-    files.set(currentFile, input)
+    if (currentFile) {
+      var input = editor.get(currentFile)
+      files.set(currentFile, input)
+    }
   }
 
   var previousInput = ''
   var compileTimeout = null
   var saveTimeout = null
 
   function editorOnChange () {
+    if (!currentFile) {
```

Comment:
Seems like my comment I made from my phone yesterday didn't go through.

This shouldn't be called at all of there's no session (aka no `currentFile`).